Core/include/Acts/MagneticField/MagneticFieldProvider.hpp

```diff
@@ -24,7 +24,7 @@ class MagneticFieldProvider {
   /// Opaque cache type that can store arbitrary implementation specific cache
   /// data. Examples are an interpolation cell, or an experiment specific
   /// conditions data handle.
-  using Cache = detail::SmallObjectCache;
+  using Cache = Acts::AnyBase<sizeof(char) * 512>;
```

Comment:
You probably just ported from `SmallObjectedCache.hpp`, but do we know why we have chosen a size of `512`?

Reply:
I wanted it to be large enough so that our interpolated field map cell fits inside and stays on the stack. The previous implementation did not overflow onto the heap, this one can. I don't think I measured the performance impact of other sizes.